### ISSUE TITLE
handling API call that returns no records

### ISSUE THREAD

**Issue by aaarendt** [communitysnowobs/mountainhub-api#8]
When extracting for this time period:
```
obs = mh.snow_data(limit=1000, start=datetime(2019,11,1), end=datetime(2019,11,3), bbox=region)
```

I get the following error:

```
------------------------------------
KeyErrorTraceback (most recent call last)
<ipython-input-22-193efdb44789> in <module>
      7 }
      8 
----> 9 obs = mh.snow_data(limit=1000, start=datetime(2019,11,1), end=datetime(2019,11,3), bbox=region)

~/miniconda3/envs/CSO-validation/lib/python3.9/site-packages/mtnhubsnow/MountainHubAPI.py in snow_data(publisher, obs_type, limit, start, end, bbox, filter)
    139     df = pd.DataFrame.from_records(parsed)
    140     if filter:
--> 141         df = df.dropna(subset=['snow_depth'])
    142     return df

~/miniconda3/envs/CSO-validation/lib/python3.9/site-packages/pandas/util/_decorators.py in wrapper(*args, **kwargs)
    309                     stacklevel=stacklevel,
    310                 )
--> 311             return func(*args, **kwargs)
    312 
    313         return wrapper

~/miniconda3/envs/CSO-validation/lib/python3.9/site-packages/pandas/core/frame.py in dropna(self, axis, how, thresh, subset, inplace)
   5946             check = indices == -1
   5947             if check.any():
-> 5948                 raise KeyError(list(np.compress(check, subset)))
   5949             agg_obj = self.take(indices, axis=agg_axis)
   5950 

KeyError: ['snow_depth']
```

I think the snow_data function might need additional checks for operating on an empty data frame? The error goes away when I broaden the time range into a period that has CSO data collected.

**Comment by emiliom** [comment]
Thanks for reporting this error! We definitely need to have an error check for this situation. I'll add it soon.

**Comment by emiliom** [comment]
Addressed in #11
